src/channel/mod.rs

```diff
@@ -11,4 +18,101 @@ trait Channel {
     fn random_number(&mut self, bound: u64) -> u64;
 
     fn random_field(&mut self) -> Self::Field;
+
+    /// Only relevant for non-interactive channels. Changes the channel seed to a "safer" seed.
+    /// 
+    /// This function guarantees that randomness fetched from the channel after calling this function
+    /// and before sending data from the prover to the channel, is "safe": A malicious
+    /// prover will have to perform 2^security_bits operations for each attempt to randomize the fetched
+    /// randomness.
+    /// 
+    /// Increases the amount of work a malicious prover needs to perform, in order to fake a proof.
+    #[allow(unused_variables)]
+    fn apply_proof_of_work(&mut self, security_bits: usize) -> Option<()> {
+        None
+    }
+
+    fn begin_query_phase(&mut self) {
+        self.get_state_mut().is_query_phase = true;
+    }
+
+    /// Channel statistics related methods
+    fn get_state(&self) -> &ChannelStates;
+    /// XXX : dunno if this is the right approach
+    fn get_state_mut(&mut self) -> &mut ChannelStates;
```

Comment:
Expose mutational getter in trait, implement other default methods depending on it

Reply:
```
get_state(&self)
```

And

```
get_state_mut(&mut self)
```

Could be `AsRef` and `AsMut` implementations: https://doc.rust-lang.org/std/convert/trait.AsRef.html